### ISSUE TITLE
Doesn't work with metapackages

### ISSUE THREAD

**Issue by aPonza** [ethz-asl/eclipsify#38]
A typical metapackage structure is:
metapackage
--package1
----src
------foo.cpp
------bar.cpp
----CMakelists.txt
----package.xml
 (...)
--packageN
----src
------baz.cpp
----CMakelists.txt
----package.xml
--metapackage
----CMakelists.txt
----package.xml

Eclipsify doesn't work in this case though, it uses the innermost metapackage folder to build the project instead of the outermost, and the output is:
```
ap@ap-NUC7i7DNKE:~/catkin_ws$ eclipsify -f metapackage -O eclipse_projects/metapackage
-- Searching workspaces:
--> Found /home/ap/catkin_ws/devel_cb, /home/ap/catkin_ws/devel, /opt/ros/kinetic

*************** Searching for package metapackage ***********************
-- Found metapackage in workspace /home/ap/catkin_ws/devel_cb with source /home/ap/catkin_ws/src/metapackage/metapackage
-- Package ambiguous also in workspace /home/ap/catkin_ws/devel with source /home/ap/catkin_ws/src/metapackage/metapackage
--> Found /home/ap/catkin_ws/src/metapackage/metapackage

eclipsify package led_lamp_testing for platform linux2
----------
-- Generating project metapackage in directory 'eclipse_projects/metapackage'
--- Overwriting eclipse_projects/metapackage/.project
--> OK
--- Overwriting eclipse_projects/metapackage/.cproject
--> OK
--- Overwriting eclipse_projects/metapackage/.settings/language.settings.xml
--> OK
-> Successfully created the project metapackage in directory eclipse_projects/metapackage
```

**Comment by HannesSommer** [comment]
Thanks, @aPonza , for reporting. Yes, this is a limitation currently. What about creating configurations for package1 to packageN ? Does that work? 

The current templates are not suited for creating a single eclipse project for all the contained packages. And so is probably the whole simplistic approach that eclipsify uses. 

If the individual packages work, then maybe the right behaviour for a metapackage would be to create individual configurations for all sub-packages.

**Comment by aPonza** [comment]
I was trying to test it and I think I broke something as now when I Ctrl+B in Eclipse I get 

<pre>
17:27:42 **** Incremental Build of configuration Default for project my_project ****
catkin build --force-color --no-status -v --no-deps --this -j8  
[build] Error: Unable to find source space `/home/user/catkin_ws/src/src`

17:27:42 Build Finished. 0 errors, 0 warnings. (took 638ms)
</pre>

Do you have any idea how to fix this? I tried recreating catkin_ws from scratch even, but nothing seems to work. I tlooked through the options for C++ Build for the project, and the usual CDT Built in Compiler Settings but nothing seems off at first glance.

**Comment by HannesSommer** [comment]
Hm, this does not look eclipse related. What happens if you execute `catkin build --force-color --no-status -v --no-deps --this -j8` directly in my_project in the catkin_ws/src/.../my_project (or similar)?

**Comment by aPonza** [comment]
The same exact output from eclipse comes out

**Comment by aPonza** [comment]
You led me to it, there was a hidden .catkin_tools folder inside src. Thanks! Tomorrow I'll update with the result of the metapackage separate build.

**Comment by HannesSommer** [comment]
That probably means your catkin_ws configuration is broken. What does `catkin config` say if run below `catkin_ws/src` ? Probably you accidentally initialized a catkin workspace in the catkin_ws/src . 
I recommend a `ls -a /home/user/catkin_ws/src/.??*` . If there are catkin related dot-files, then you should probably delete them. Similar with potential devel, build folders below src/.

**Comment by aPonza** [comment]
I tried again and created a barebones metapackage with the structure I listed: after setting the exec depends on the metapackage to the other 2 packages, if one `catkin build`s the metapackage then all 3 contained packages are eclipsify-able and seem to build correctly like you suggested. Thank you!

It took quite a while to fix the index however, as Eclipse again didn't want to recognize the ros includes, but this time it seemed more stubborn than ever. What did it was following [this](https://github.com/ethz-asl/programming_guidelines/wiki/Catkin-Eclipse#preliminary-setup-global-settings-in-eclipse) and setting the "global provider shared between projects in both output parser and built-in compiler" for all the projects (i.e. packages in the metapackage) as per [this](https://github.com/ethz-asl/programming_guidelines/wiki/Catkin-Eclipse#adding-a-catkin-package-as-project-to-eclipse).

Thanks for the great generator and related support.
